### PR TITLE
Implement ensure_resources and spec tests

### PR DESCRIPTION
**Pull request by mleiner**

### CODE REVIEW

File: data/common.yaml [hunk 1]
```diff
@@ -17,21 +18,19 @@ exim::log_tls_peerdn: true
 exim::macros: {}
 exim::manage_service: false
 exim::never_users: [ 'root' ]
-exim::mysql_servers: ~
```
Comment (mleiner) [inline data/common.yaml:20]: sorted the file!

File: data/spec_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+exim::retry:
+  'ensureresource': {}
```
Comment (mleiner) [inline data/spec_tests.yaml:3]: needed for tests

File: spec/hiera.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (mleiner) [inline spec/hiera.yaml:1]: Copy of hiera.yaml to spec/hiera.yaml PLUS spec_tests.yaml in hierarchy for testing

File: spec/spec_helper_local.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+RSpec.configure do |c|
```
Comment (mleiner) [inline spec/spec_helper_local.rb:1]: Making use of the custom hiera.yaml

**Review by jvetter83 — APPROVED** [review]

**Pull request merged**